test/letters/containers/Main.unit.spec.jsx

```diff
@@ -1,55 +1,134 @@
+    expect(childText).to.equal(testText);
   });
 
-  it('should show loading spinner when waiting for response', () => {
-    const props = _.merge({}, defaultProps, { lettersAvailability: 'awaitingResponse' });
+  it('shows a loading spinner when awaiting response', () => {
+    const props = { ...defaultProps, lettersAvailability: 'awaitingResponse' };
     const tree = SkinDeep.shallowRender(<Main {...props}/>);
     expect(tree.subTree('LoadingIndicator')).to.be.ok;
   });
 
-  it('should show system down message for backend service error', () => {
-    const props = _.merge({}, defaultProps, { lettersAvailability: 'backendServiceError' });
+  it('shows a system down message for backend service error', () => {
+    const props = { ...defaultProps, lettersAvailability: 'backendServiceError' };
     const tree = SkinDeep.shallowRender(<Main {...props}/>);
     expect(tree.subTree('#systemDownMessage')).to.be.ok;
   });
 
   it('should show backend authentication error', () => {
-    const props = _.merge({}, defaultProps, { lettersAvailability: 'backendAuthenticationError' });
+    const props = { ...defaultProps, lettersAvailability: 'backendAuthenticationError' };
     const tree = SkinDeep.shallowRender(<Main {...props}/>);
     expect(tree.subTree('#recordNotFound')).to.be.ok;
   });
 
   it('should show system down message for invalid address error', () => {
-    const props = _.merge({}, defaultProps, { lettersAvailability: 'invalidAddressProperty' });
+    const props = { ...defaultProps, lettersAvailability: 'invalidAddressProperty' };
     const tree = SkinDeep.shallowRender(<Main {...props}/>);
     expect(tree.subTree('#systemDownMessage')).to.be.ok;
   });
 
+  it('renders children for letter eligibility errors', () => {
+    const props = { ...defaultProps, lettersAvailability: 'letterEligibilityError', children };
+    const tree = SkinDeep.shallowRender(<Main {...props}/>);
+    const childText = tree.subTree('span').text();
+    expect(childText).to.equal(testText);
+  });
+
   it('should show letters unavailable message when service is unavailable', () => {
-    const props = _.merge({}, defaultProps, { lettersAvailability: 'unavailable' });
+    const props = { ...defaultProps, lettersAvailability: 'unavailable' };
     const tree = SkinDeep.shallowRender(<Main {...props}/>);
     expect(tree.subTree('#lettersUnavailable')).to.be.ok;
   });
+
+  it('renders system down message for all unspecified errors', () => {
+    const props = { ...defaultProps, lettersAvailability: 'bogusError' };
+    const tree = SkinDeep.shallowRender(<Main {...props}/>);
+    expect(tree.subTree('#systemDownMessage')).to.be.ok;
+  });
+
+  it('fetches letter data after mounting', () => {
+    const getLetterList = sinon.spy();
+    const props = { ...defaultProps, getLetterList };
+    const tree = SkinDeep.shallowRender(<Main {...props}/>);
+    const instance = tree.getMountedInstance();
+    instance.componentDidMount();
```

Comment:
Did you check that you had to call this explicitly? Seems weird that `getMountedInstance()` wouldn't go through these mounting lifecycle methods, but I wouldn't be surprised. Just wondering if you checked.

Probably wouldn't hurt to put in a comment explaining _why_ you had to call this manually for anybody who looks later and perhaps isn't familiar with how SkinDeep does things.

Reply:
Yeah, I couldn't find great documentation on this, but from what I can tell in testing, you need to call this lifecycle method explicitly. I think it might vary from method to method, though. Will add comment.